### PR TITLE
Moved the import and export buttons to the top of the canvas

### PR DESCRIPTION
**Pull request by anthonydillon**

### CODE REVIEW

**Comment by anthonydillon** [comment]
## QA
- Make sure the import export buttons are now at the top left next to the environment name
- Check it matches the design [1]
- Add a service and see that the export button becomes visible
- Export a bundle and check it downloads the yaml correctly
- Import a bundle and check it works

[1] https://cloud.githubusercontent.com/assets/9105999/14288312/01d7e9de-fb4e-11e5-9892-3ee3f0847ac9.png

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4921/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4923/
Test PASSed.

**Comment by bac** [comment]
QA notes:

* Your implementation has import / export icons.  Design has the reverse.
* Initially only 'import' is shown. After adding a service, 'export' is shown. However if all services are deleted the 'export' button is still shown. Destroying all machines makes it go away. In the case where there is a machine but no services, the export produces:
```
series: trusty
services: {}
relations: []
machines: {}
```
Since that bundle is pretty useless, perhaps the export button should be removed/disabled when there are no services.

**Comment by Lucapaulina** [comment]
@bac I think export is broken here. Users should be able to model an environment and then export it without having to commit the model.

**Comment by anthonydillon** [comment]
@bac thanks for the feedback. I've fixed the ordering but from @Lucapaulina comment it seems the export button is appearing correctly but the exported bundle should contain uncommited changes.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4929/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4930/
Test PASSed.

**Comment by anthonydillon** [comment]
@bac This is ready for re-review.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4935/
Test PASSed.

**Comment by bac** [comment]
@Lucapaulina and @anthonydillon I never said anything about uncommitted changes.  The state I described was committed.

Looky here, I deployed Hadoop and committed everything.  Export button is now visible. I then destroy Hadoop *and* commit.  (See below) No services but still one machine.

screen shot 2016-04-29 at 13 13 49

The export still works but only produces
```
series: trusty
services: {}
relations: []
machines: {}
```

I'm just saying if you're going to completely take away the button then you should do it when there are zero services left not wait until there are zero machines.

You didn't address my question about removing the button vs. disabling it so I assume @Lucapaulina is happy with what you've done.

The buttons are in the right order. I still have the question above but am happy to call this QA-OK but hope you'll give some thought to what I've asked.

File: jujugui/static/gui/src/app/components/import-export/import-export.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+    /**
+      When file is submitted the drag over animation is triggered and the file
+      is passed to the utils function.
+
+      @method _handleImportFile
+    */
+    _handleImportFile: function() {
+      var inputFile = this.refs['file-input'].files[0];
+      if(inputFile) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/components/import-export/import-export.js]: `if (`

File: jujugui/templates/index.html.mako [hunk 2]
```diff
@@ -88,6 +88,7 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
           % endif
         </div>
         <div id="header-breadcrumb"></div>
+        <div id="import-export-container"></div>
```
Comment (hatched) [inline jujugui/templates/index.html.mako]: This also needs to be added to index.html.go

**Comment by hatched** [comment]
Code looks good but the container element will need to be added to `index.html.go` and then qa'd in Juju 2

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4938/
Test PASSed.

**Comment by hatched** [comment]
Thanks for adding that!

**Comment by anthonydillon** [comment]
@bac Thanks for the feedback. I agree there is no point in enabling/showing the export when there is nothing to export. I will work to add this functionality. I also agree disabling the button is better then hiding it but that is how it was so I guessed it was thought about. 

@Lucapaulina what do you think? Hide the import button or disable it?

**Comment by bac** [comment]
Thanks for the late-night pre-bank holiday fix @anthonydillon .  doing QA now.

**Comment by bac** [comment]
Works in GIJoe!  Thanks.

**Comment by anthonydillon** [comment]
I think this extra functionally maybe outside of this branch as it's
awaiting design feedback. I have created an issue on the design project to
get formal feedback.

https://github.com/CanonicalLtd/juju-design/issues/78

In this case, I think this can land as a styling branch and the result of
the design issue can be a functional update. Feel free to land if I'm not
around.

**Comment by hatched** [comment]
Thanks @anthonydillon could you rebase this to remove the conflict.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4943/
Test PASSed.

**Comment by makyo** [comment]
:+1:

**Comment by hatched** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**